tests/utils-e2e.ts

```diff
@@ -394,207 +394,108 @@ export async function testE2E(
   replaceTenderlyWithEstimateGas?: boolean,
   forceRoute?: AddressOrSymbol[],
 ) {
+  const useAPI = testingEndpoint && !poolIdentifiers;
+  // The API currently doesn't allow for specifying poolIdentifiers
+  const sdk: IParaSwapSDK = useAPI
+    ? new APIParaswapSDK(network, dexKeys, '')
+    : new LocalParaswapSDK(network, dexKeys, '', limitOrderProvider);
+  // initialize pricing
+  await sdk.initializePricing?.();
+  // fetch the route
   const amount = BigInt(_amount);
-
-  const ts: TransactionSimulator = replaceTenderlyWithEstimateGas
-    ? new EstimateGasSimulation(new DummyDexHelper(network).provider)
-    : new TenderlySimulation(network);
-  await ts.setup();
-
-  if (srcToken.address.toLowerCase() !== ETHER_ADDRESS.toLowerCase()) {
-    // check if v5 is available in the config
-    if (generateConfig(network).tokenTransferProxyAddress !== NULL_ADDRESS) {
-      const allowanceTx = await ts.simulate(
-        allowTokenTransferProxyParams(srcToken.address, senderAddress, network),
-      );
-      if (!allowanceTx.success) console.log(allowanceTx.url);
-      expect(allowanceTx!.success).toEqual(true);
-    }
-    const augustusV6Allowance = await ts.simulate(
-      allowAugustusV6(srcToken.address, senderAddress, network),
-    );
-    if (!augustusV6Allowance.success) console.log(augustusV6Allowance.url);
-    expect(augustusV6Allowance!.success).toEqual(true);
-  }
-
-  if (deployedTestContractAddress) {
-    const whitelistTx = await ts.simulate(
-      augustusGrantRoleParams(
-        deployedTestContractAddress,
-        network,
-        testContractType || 'adapter',
-      ),
-    );
-    expect(whitelistTx.success).toEqual(true);
-    console.log(`Successfully whitelisted ${deployedTestContractAddress}`);
-
-    if (testContractType === 'router') {
-      const setImplementationTx = await ts.simulate(
-        augustusSetImplementationParams(
-          deployedTestContractAddress,
-          network,
-          contractMethod,
-        ),
-      );
-      expect(setImplementationTx.success).toEqual(true);
-    }
-  } else if (testContractBytecode) {
-    const deployTx = await ts.simulate(
-      deployContractParams(testContractBytecode, network),
+  const priceRoute = await sdk.getPrices(
+    srcToken,
+    destToken,
+    amount,
+    swapSide,
+    contractMethod,
+    poolIdentifiers,
+    transferFees,
+    forceRoute,
+  );
+  // log the route for visibility
+  console.log('Price Route:', JSON.stringify(priceRoute, null, 2));
+  // prepare state overrides
+  const tenderlySimulator = TenderlySimulatorNew.getInstance();
+  // any address works
+  const userAddress = TenderlySimulatorNew.DEFAULT_OWNER;
+  // init `StateOverride` object
+  const stateOverride: StateOverride = {};
+  // fund x2 just in case
+  const amountToFund = amount * 2n;
```

Comment:
this is not going to work for `BUY`, check `BUY` tests with something like ETH -> USDC